enhancements/scorecard-storage.md

```diff
@@ -0,0 +1,333 @@
+to perform data management of this test output directory as they require.
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, or as something else. Initial proposal
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this
+enhancement:
+- Maturity levels - `Dev Preview`, `Tech Preview`, `GA`
+- Deprecation
+
+Clearly define what graduation means.
+
+#### Examples
+
+These are generalized examples to consider, in addition to the aforementioned
+[maturity levels][maturity-levels].
+
+##### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+##### Tech Preview -> GA 
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+
+
+### Upgrade / Downgrade Strategy
+
+This feature doesn't impact existing scorecard tests or test configurations.
+
+I currently don't see value in backporting this feature to previous versions of the operator-sdk.
+
+### Version Skew Strategy
+
+
+## Implementation History
+
+
+
+## Drawbacks
+
+This feature and suggested implementation depend on dynamic storage
+being available on-cluster.
+
+
+## Alternatives
+
+### Alternative 1
+An alternative to this feature would be to design a means of embedding test content into the test Pod log itself, co-mingled with the current scorecard test result JSON.
+
+This seems really complicated and hard to maintain so I rejected this in favor of this proposal.
+
+### Alternative 2
+
+Another alternative to creating PVCs for test output storage, you could:
+ * create a sidecar container in the test pod
+ * share an emptydir volume between the containers
+ * have the test pod write to that emptydir any test output
+ * change scorecard to not remove the test pod until the
+test output is harvested from the sidecar.
+
+This would work, but it means depending on emptyDir storage which makes
+per-test configuration of storage probably not possible.  The other
```

Comment:
it doesn't depend on emptydir storage, it makes emptydir storage an option.  You can still use PVs if you really wanted to (if you chose to expose that config option).

that said, i'd like to hear why per-test configuration of storage is critically important.

we could start by building the emptydir storage capability and, if needed, extend it to allow allocation of PVs also.

Reply:
not all tests will require storage, some will and other not, so being able to specify this on a per test basis seems to make sense to me.  Having unique storage for each test insures tests don't overwrite results since tests run in parallel and multiple tests could specify the same mountpaths.

yes, we could scrap this design and go with the alternative design (emptydir/sidecar) if that is the direction people want to go.